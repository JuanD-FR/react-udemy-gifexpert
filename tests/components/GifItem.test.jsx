import { render, screen } from "@testing-library/react";
import { GifItem } from "../../src/components/GifItem";

describe('Pruebas en <GiftItem />', () => {
    const title='Super smash';
    const url='https://supersmashbros.com/link.jpg';

    test('Debe hacer match con el snapshot', () => { 
            const { container } = render(<GifItem title={title} url={url} />);
            expect(container).toMatchSnapshot();
         });

    test('debe mostrar imagen con URL y ALT indicados', () => { 
        
        render(<GifItem title={title} url={url} />);
        const {src, alt} = screen.getByRole('img');
        expect(src).toBe(url);
        expect(alt).toBe(title);
     });

     test('debe mostrar el titulo en el componente', () => { 
        
        render(<GifItem title={title} url={url} />);
        expect(screen.getByText(title)).toBeTruthy();
        
     });
});