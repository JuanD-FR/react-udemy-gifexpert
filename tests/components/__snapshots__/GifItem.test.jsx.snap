// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GiftItem /> Debe hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Super smash"
      src="https://supersmashbros.com/link.jpg"
    />
    <p>
      Super smash
    </p>
  </div>
</div>
`;
